File: SerpentineApi/SerpentineApi/Responses/BaseResponse.cs
namespace SerpentineApi.Responses;

public abstract class BaseResponse
{
    public Ulid Id { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public string CreatedAtIso => CreatedAt.ToString("yyyy-MM-ddTHH:mm:ss.fffffffZ");
    public string UpdatedAtIso => UpdatedAt.ToString("yyyy-MM-ddTHH:mm:ss.fffffffZ");
}

File: serpentineclient/src/models/responses/base-response.ts
export abstract class BaseResponse {
    id: string = "";
    createdAt: string = "";
    updatedAt: string = "";
    createdAtIso: string = "";
    updatedAtIso: string = "";

    constructor(init?: Partial<BaseResponse>) {
    Object.assign(this, init);
  }
  
}